### PR TITLE
Avoid Guice circular issue and implement test to ensure extension is activated

### PR DESCRIPTION
**Pull request by jonesbusy**
### Testing done

Automated tests

```[tasklist]
### Submitter checklist
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue
```

### CODE REVIEW

**Comment by jonesbusy** [comment]
Can you take a look @sridamul

File: src/test/java/org/jenkinsci/plugins/ConfigurableExtensionFilterTest.java [hunk 1]
```diff
@@ -2,36 +2,67 @@
...
 public class ConfigurableExtensionFilterTest {
 
-    @Rule
-    public JenkinsRule j = new JenkinsRule();
-
     @Test
-    public void nullSafety() {
-        DescriptorImpl impl = new DescriptorImpl();
+    public void nullSafety(JenkinsRule j) {
+        ConfigurableExtensionFilter.DescriptorImpl impl =
```
Comment (jonesbusy) [inline src/test/java/org/jenkinsci/plugins/ConfigurableExtensionFilterTest.java:20]: If we use JenkinsRule we need to check with the desciptor from JenkinsRule and not an object created directly in the test

**Comment by sridamul** [comment]
Can you also remove the validation? or need to figure out other way to display the message correctly.

**Comment by sridamul** [comment]
Can we use the extension from the extension Indexer? If possible, we can use it after the completion of [this](https://www.jenkins.io/projects/gsoc/2024/project-ideas/fixing-backend-extension-indexer-tool/) GSoC'24 project

**Comment by jonesbusy** [comment]
> Can you also remove the validation? or need to figure out other way to display the message correctly.

https://github.com/jenkinsci/extension-filter-plugin/pull/70

By reading doc this can be achieved using an other class loader to check class existence

**Comment by jonesbusy** [comment]
> Can we use the extension from the extension Indexer to validate class? If possible, we can use it after the completion of [this](https://www.jenkins.io/projects/gsoc/2024/project-ideas/fixing-backend-extension-indexer-tool/) GSoC'24 project

Not sure what you mean. But we are talking about 2 different tool here (One plugin that can disable extension on a running controller and one tool that just display the available extension across all plugin)

**Comment by sridamul** [comment]
> Not sure what you mean. But we are talking about 2 different tool here (One plugin that can disable extension on a running controller and one tool that just display the available extension across all plugin)

I thought that one can check whether the class names are available in the extension indexer. If it presents, then it is an extension and can be indicated with correct validation



As you suggested 
> By reading doc this can be achieved using an other class loader to check class existence

this is better approach.

**Comment by sridamul** [comment]
> Can you take a look @sridamul

tested the code locally (both descriptor and extension)
LGTM!

**Comment by sridamul** [comment]
If possible can you remove the `default="&{null}"`in the https://github.com/jenkinsci/extension-filter-plugin/blob/main/src/main/resources/org/jenkinsci/plugins/ConfigurableExtensionFilter/global.jelly#L12
 as suggestion by 'Daniel Beck'

**Pull request merged**